From cc873049ff9249a33567e4cbcdeffeb6806a30d6 Mon Sep 17 00:00:00 2001
From: brianlheim <self@brianlheim.com>
Date: Tue, 25 Aug 2020 17:14:29 -0500
Subject: [PATCH] remove workarounds for old boost versions

we don't need to support boost pre-1.70 anymore, and tcp::socket::executor_type was always the right typealias to use
---
 server/supernova/sc/sc_osc_handler.hpp | 10 +---------
 1 file changed, 1 insertion(+), 9 deletions(-)

diff --git a/server/supernova/sc/sc_osc_handler.hpp b/server/supernova/sc/sc_osc_handler.hpp
index f17e99a5573..676c94dbf15 100644
--- a/server/supernova/sc/sc_osc_handler.hpp
+++ b/server/supernova/sc/sc_osc_handler.hpp
@@ -234,11 +234,7 @@ class sc_osc_handler : private detail::network_thread, public sc_notify_observer
     class tcp_connection : public nova_endpoint {
     public:
         using pointer = std::shared_ptr<tcp_connection>;
-#if BOOST_VERSION >= 107000
-        using executor = boost::asio::executor;
-#else
-        using executor = boost::asio::io_context::executor_type;
-#endif
+        using executor = tcp::socket::executor_type;
 
         static pointer create(const executor& executor) { return pointer(new tcp_connection(executor)); }
 
@@ -249,11 +245,7 @@ class sc_osc_handler : private detail::network_thread, public sc_notify_observer
         bool operator==(tcp_connection const& rhs) const { return &rhs == this; }
 
     private:
-#if BOOST_VERSION >= 107000
         tcp_connection(const executor& executor): socket_(executor) {}
-#else
-        tcp_connection(const executor& executor): socket_(executor.context()) {}
-#endif
 
         void send(const char* data, size_t length) override final;
 
